.github/actions/deploy/action.yml

```diff
@@ -0,0 +1,195 @@
+    - name: Set Environment variables
+      id: set_env_var
+      shell: bash
+      run: |
+        tf_vars_file=terraform/workspace_variables/${{ inputs.environment }}.tfvars.json
+
+        if [ -n "${{ inputs.pr }}" ]; then
+          echo "DEPLOY_ENV=review-${{ inputs.pr }}" >> $GITHUB_ENV
+          echo "DEPLOY_URL=https://apply-review-${{ inputs.pr }}.london.cloudapps.digital" >> $GITHUB_ENV
+        else
+          echo "DEPLOY_ENV=${{ inputs.environment }}" >> $GITHUB_ENV
+
+          hostname=$(jq -r '.service_gov_uk_host_names[0]' ${tf_vars_file})
+          if [[ $hostname != null ]]; then
+            echo "DEPLOY_URL=https://${hostname}.apply-for-teacher-training.service.gov.uk" >> $GITHUB_ENV
+          else
+            echo "DEPLOY_URL=https://apply-${{ inputs.environment }}.london.cloudapps.digital" >> $GITHUB_ENV
+          fi
+        fi;
+        echo "DOCKER_IMAGE=$DOCKER_IMAGE" >> $GITHUB_ENV
+
+        echo "KEY_VAULT_NAME=$(jq -r '.key_vault_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "KEY_VAULT_APP_SECRET_NAME=$(jq -r '.key_vault_app_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "KEY_VAULT_INFRA_SECRET_NAME=$(jq -r '.key_vault_infra_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "KEY_VAULT_CYPRESS_SECRET_NAME=$(jq -r '.key_vault_cypress_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "PAAS_CF_SPACE=$(jq -r '.paas_cf_space' ${tf_vars_file})" >> $GITHUB_ENV
+      env:
+        DOCKER_IMAGE: ${{ format('ghcr.io/dfe-digital/apply-teacher-training:{0}', inputs.sha) }}
+
+    - name: Start ${{ env.DEPLOY_ENV }} Deployment
+      if: ${{ always() }}
+      uses: bobheadxi/deployments@v1
+      id: deployment
+      with:
+        env:   ${{ env.DEPLOY_ENV }}
+        step:  start
+        token: ${{ inputs.actions-api-access-token }}
+
+    - name: Use Terraform v0.14.9
+      id: use_terraform
+      uses: hashicorp/setup-terraform@v1
+      with:
+        terraform_version: 0.14.9
+
+    - uses: azure/login@v1
+      id: login_azure
+      with:
+        creds: ${{ inputs.azure-credentials }}
+
+    - uses: DFE-Digital/keyvault-yaml-secret@v1
+      id: get_secrets
+      with:
+        keyvault: ${{ env.KEY_VAULT_NAME }}
+        secret: ${{ env.KEY_VAULT_INFRA_SECRET_NAME }}
+        key: CF_USER, CF_PASSWORD
+
+    - uses: DFE-Digital/keyvault-yaml-secret@v1
+      id: get_cypress_secrets
+      with:
+        keyvault: ${{ env.KEY_VAULT_NAME }}
+        secret: ${{ env.KEY_VAULT_CYPRESS_SECRET_NAME }}
+        key: CANDIDATE_TEST_EMAIL, GOVUK_NOTIFY_API_KEY
+
+    - name: Setup cf cli
+      id: setup_cf_cli
+      uses: DFE-Digital/github-actions/setup-cf-cli@master
+      with:
+        CF_USERNAME:   ${{ steps.get_secrets.outputs.cf_user }}
+        CF_PASSWORD:   ${{ steps.get_secrets.outputs.cf_password }}
+        CF_SPACE_NAME: ${{ env.PAAS_CF_SPACE }}
+        INSTALL_CONDUIT: true
+
+    - name: Validate Azure Key Vault secrets
```

Comment:
Can this step be moved up or down? So that setup cf cli is next to validate KV where it's used